### ISSUE TITLE
Not working since v3.2.5 (x86 only)

### ISSUE THREAD

**Issue by KamYEpiC** [SpriteOvO/Telegram-Anti-Revoke#88]
[https://pasteboard.co/EWUoTAva3jK2.jpg](https://pasteboard.co/EWUoTAva3jK2.jpg)
Hi, Crash with a new telegram update, thanks for your support

**Comment by SpriteOvO** [comment]
Can you upload the content of `ArLog.txt` (in the Telegram directory)?

**Comment by SuChenUp** [comment]
```
[2021-11-17 15:40:01.494] [Main] [info] [RealMain.cpp:32] Running. Version: "0.4.2", Platform: "x64"
[2021-11-17 15:40:01.496] [Main] [info] [IRuntime.cpp:27] [IRuntime] Telegram version: 1000001
[2021-11-17 15:40:07.989] [Main] [info] [IUpdater.cpp:263] [Updater] Get data by bridge successed.
[2021-11-17 15:40:07.990] [Main] [info] [IUpdater.cpp:107] [Updater] No need to update. Local: 000004002, Latest: 000004002
[2021-11-17 15:40:07.990] [Main] [info] [IUpdater.cpp:39] [Updater] ParseResponse() successed. (ByBridge)
[2021-11-17 15:40:08.190] [Main] [info] [IRuntime.cpp:125] [IRuntime] InitDynamicData_MallocFree() succeeded.
[2021-11-17 15:40:08.350] [Main] [info] [IRuntime.cpp:126] [IRuntime] InitDynamicData_DestroyMessage() succeeded.
[2021-11-17 15:40:08.501] [Main] [warning] [IRuntime.cpp:676] [IRuntime] Search EditedIndex failed.
[2021-11-17 15:40:08.501] [Main] [warning] [IRuntime.cpp:127] [IRuntime] InitDynamicData_EditedIndex() failed.
[2021-11-17 15:40:08.501] [Main] [critical] [RealMain.cpp:51] [IRuntime] InitDynamicData() failed.
```

I also encountered this problem in the process of using it. This is the content in ArLog.txt, I hope it can help you solve the problem.

**Comment by SpriteOvO** [comment]
@SuChenUp The logs are a bit strange, the plugin does not get the correct Telegram version number and does not trigger an under version check:
https://github.com/SpriteOvO/Telegram-Anti-Revoke/blob/985a2eca33fdfbde5a4dc4ed3e398cf174af8c6d/Source/Core/IRuntime.cpp#L36-L38

Are you using a third-party Telegram client or is the plugin you are using modified by yourself or someone else?

**Comment by SuChenUp** [comment]
@SpriteOvO Sorry, I forgot that I am using a third-party Telegram client, but everything works fine when I use the plug-in in Telegram 3.2.2.

**Comment by Gigabait** [comment]
Today i get the same problem.

```
[2021-11-17 13:11:16.986] [Main] [info] [RealMain.cpp:32] Running. Version: "0.4.2", Platform: "x86"
[2021-11-17 13:11:16.988] [Main] [info] [IRuntime.cpp:27] [IRuntime] Telegram version: 3002005
[2021-11-17 13:11:50.044] [Main] [warning] [IUpdater.cpp:244] [Updater] Internet::HttpRequest() failed. (ByBridge)
[2021-11-17 13:11:50.044] [Main] [warning] [IUpdater.cpp:35] [Updater] GetDataByBridge() failed, try GetDataDirectly().
[2021-11-17 13:11:50.580] [Main] [info] [IUpdater.cpp:294] [Updater] Get data directly successed.
[2021-11-17 13:11:50.580] [Main] [info] [IUpdater.cpp:107] [Updater] No need to update. Local: 000004002, Latest: 000004002
[2021-11-17 13:11:50.580] [Main] [info] [IUpdater.cpp:56] [Updater] ParseResponse() successed. (Directly)
[2021-11-17 13:11:50.714] [Main] [info] [IRuntime.cpp:125] [IRuntime] InitDynamicData_MallocFree() succeeded.
[2021-11-17 13:11:50.770] [Main] [info] [IRuntime.cpp:126] [IRuntime] InitDynamicData_DestroyMessage() succeeded.
[2021-11-17 13:11:50.838] [Main] [info] [IRuntime.cpp:127] [IRuntime] InitDynamicData_EditedIndex() succeeded.
[2021-11-17 13:11:50.899] [Main] [info] [IRuntime.cpp:128] [IRuntime] InitDynamicData_SignedIndex() succeeded.
[2021-11-17 13:11:50.976] [Main] [info] [IRuntime.cpp:129] [IRuntime] InitDynamicData_ReplyIndex() succeeded.
[2021-11-17 13:11:51.062] [Main] [info] [IRuntime.cpp:130] [IRuntime] InitDynamicData_LangInstance() succeeded.
[2021-11-17 13:11:51.148] [Main] [warning] [IRuntime.cpp:1263] [IRuntime] Search toHistoryMessage index falied.
[2021-11-17 13:11:51.148] [Main] [warning] [IRuntime.cpp:131] [IRuntime] InitDynamicData_ToHistoryMessage() failed.
[2021-11-17 13:11:51.148] [Main] [critical] [RealMain.cpp:51] [IRuntime] InitDynamicData() failed.
```

**Comment by hitokirihahn** [comment]
Same problem 

`[2021-11-17 15:45:20.243] [Main] [info] [RealMain.cpp:32] Running. Version: "0.4.2", Platform: "x86"
[2021-11-17 15:45:20.245] [Main] [info] [IRuntime.cpp:27] [IRuntime] Telegram version: 3002005
[2021-11-17 15:45:22.083] [Main] [info] [IUpdater.cpp:263] [Updater] Get data by bridge successed.
[2021-11-17 15:45:22.083] [Main] [info] [IUpdater.cpp:107] [Updater] No need to update. Local: 000004002, Latest: 000004002
[2021-11-17 15:45:22.083] [Main] [info] [IUpdater.cpp:39] [Updater] ParseResponse() successed. (ByBridge)
[2021-11-17 15:45:22.306] [Main] [info] [IRuntime.cpp:125] [IRuntime] InitDynamicData_MallocFree() succeeded.
[2021-11-17 15:45:22.408] [Main] [info] [IRuntime.cpp:126] [IRuntime] InitDynamicData_DestroyMessage() succeeded.
[2021-11-17 15:45:22.522] [Main] [info] [IRuntime.cpp:127] [IRuntime] InitDynamicData_EditedIndex() succeeded.
[2021-11-17 15:45:22.629] [Main] [info] [IRuntime.cpp:128] [IRuntime] InitDynamicData_SignedIndex() succeeded.
[2021-11-17 15:45:22.759] [Main] [info] [IRuntime.cpp:129] [IRuntime] InitDynamicData_ReplyIndex() succeeded.
[2021-11-17 15:45:22.893] [Main] [info] [IRuntime.cpp:130] [IRuntime] InitDynamicData_LangInstance() succeeded.
[2021-11-17 15:45:23.029] [Main] [warning] [IRuntime.cpp:1263] [IRuntime] Search toHistoryMessage index falied.
[2021-11-17 15:45:23.029] [Main] [warning] [IRuntime.cpp:131] [IRuntime] InitDynamicData_ToHistoryMessage() failed.
[2021-11-17 15:45:23.029] [Main] [critical] [RealMain.cpp:51] [IRuntime] InitDynamicData() failed.
`

**Comment by KamYEpiC** [comment]
> Can you upload the content of `ArLog.txt` (in the Telegram directory)?

[2021-11-17 10:16:27.713] [Main] [info] [RealMain.cpp:32] Running. Version: "0.4.2", Platform: "x86"
[2021-11-17 10:16:27.715] [Main] [info] [IRuntime.cpp:27] [IRuntime] Telegram version: 3002005
[2021-11-17 10:16:30.029] [Main] [info] [IUpdater.cpp:263] [Updater] Get data by bridge successed.
[2021-11-17 10:16:30.029] [Main] [info] [IUpdater.cpp:107] [Updater] No need to update. Local: 000004002, Latest: 000004002
[2021-11-17 10:16:30.029] [Main] [info] [IUpdater.cpp:39] [Updater] ParseResponse() successed. (ByBridge)
[2021-11-17 10:16:30.376] [Main] [info] [IRuntime.cpp:125] [IRuntime] InitDynamicData_MallocFree() succeeded.
[2021-11-17 10:16:30.533] [Main] [info] [IRuntime.cpp:126] [IRuntime] InitDynamicData_DestroyMessage() succeeded.
[2021-11-17 10:16:30.706] [Main] [info] [IRuntime.cpp:127] [IRuntime] InitDynamicData_EditedIndex() succeeded.
[2021-11-17 10:16:30.877] [Main] [info] [IRuntime.cpp:128] [IRuntime] InitDynamicData_SignedIndex() succeeded.
[2021-11-17 10:16:31.071] [Main] [info] [IRuntime.cpp:129] [IRuntime] InitDynamicData_ReplyIndex() succeeded.
[2021-11-17 10:16:31.280] [Main] [info] [IRuntime.cpp:130] [IRuntime] InitDynamicData_LangInstance() succeeded.
[2021-11-17 10:16:31.490] [Main] [warning] [IRuntime.cpp:1263] [IRuntime] Search toHistoryMessage index falied.
[2021-11-17 10:16:31.490] [Main] [warning] [IRuntime.cpp:131] [IRuntime] InitDynamicData_ToHistoryMessage() failed.


THIS is mine too, I'm using the original telegram version 3.2.5

**Comment by SpriteOvO** [comment]
The x64 version still seems to work, I will try to fix the x86 version later.

**Comment by hitokirihahn** [comment]
> The x64 version still seems to work, I will try to fix the x86 version later.

i change to x64 version and it works flawlessly

**Comment by KamYEpiC** [comment]
> > The x64 version still seems to work, I will try to fix the x86 version later.
> 
> i change to x64 version and it works flawlessly

**how did you change it to x64??**

**Comment by SpriteOvO** [comment]
@KamYEpiC Download `Windows 64 bit: Portable or Installer` in https://github.com/telegramdesktop/tdesktop/releases/tag/v3.2.5

**Comment by Howard20181** [comment]
> @SpriteOvO Sorry, I forgot that I am using a third-party Telegram client, but everything works fine when I use the plug-in in Telegram 3.2.2.

Maybe you are using 64Gram, which resets the version number. https://github.com/TDesktop-x64/tdesktop/commit/b7bd942c85523b3f162b93bcaf3959587d39935e#diff-5b51bb43daa9c4ae5d387137c30984918f0310760c08246ce68b1c4eedb06573

**Comment by rhybosoma** [comment]
> > > The x64 version still seems to work, I will try to fix the x86 version later.
> > 
> > 
> > i change to x64 version and it works flawlessly
> 
> **how did you change it to x64??**

I think I had installed the official version. You may check it for yourself. 

In my case, I went to https://desktop.telegram.org/
from there installed the x64 client (default path), so I assume that it overwrote the client/program. 
I did not even need to sign in again.
Hope it helps someone.

Cheers!

**Comment by SpriteOvO** [comment]
Fixed in the [v0.4.3](https://github.com/SpriteOvO/Telegram-Anti-Revoke/releases/tag/0.4.3).

If you are a Telegram x64 user and using TAR v0.4.2, this version update is not required.
